### PR TITLE
docs: add more hints to observability docs; add database intellij docs

### PR DESCRIPTION
**Pull request by bahrmichael**
This PR enriches the observability docs from my own experience, and adds a doc on using the Database plugin from IntelliJ.

## Test plan

No tests, docs only.

### CODE REVIEW

File: doc/dev/how-to/inspect_postgres_from_intellij.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+If you haven't done so yet, run `sg setup` and `sg start` to make sure your database is up and running.
+
+Below we'll assume the default config for postgres. It is defined in [`sg.config.yaml`](/sg.config.yaml) (or your overrides),
+and has `env` keys like `PGPASSWORD` and `PGPORT`.
+
+## 1. Open the database plugin
+
+It's usually on the right edge of your IDE.
+
+![](https://github.com/sourcegraph/sourcegraph/assets/1830132/064d1970-c9d5-4605-8242-1776b76a1ec2)
```
Comment (jhchabran) [inline doc/dev/how-to/inspect_postgres_from_intellij.md:15]: It sounds these have been uploaded in a PR/Issue somewhere, they might expire unless referenced, see https://github.com/orgs/community/discussions/77006#discussioncomment-7670246 

Best to instead follow https://docs.sourcegraph.com/dev/how-to/documentation_implementation#adding-images-to-the-documentation if you still plan to merge this.

Comment (bahrmichael) [reply doc/dev/how-to/inspect_postgres_from_intellij.md:15]: Will update that, thanks for the pointer :)

**Review by jhchabran — CHANGES_REQUESTED** [review]
As much as I want to merge this, the images you're bringing here are above 100kb, so they should be instead stored in GCS, cf the link I shared: 

CleanShot 2024-04-29 at 11 06 26@2x

**Comment by bahrmichael** [comment]
@jhchabran Sorry, I failed to review the image sizes. I've updated it now, and moved all images together. Even though some are smaller, I think it's better to keep the group of images together.

**Comment by jhchabran** [comment]
> @jhchabran Sorry, I failed to review the image sizes. 

Don't worry, this is a bit counter intuitive and it would be best if we had a CI check that spotted this. 

> I've updated it now, and moved all images together. Even though some are smaller, I think it's better to keep the group of images together.

Fully agreeing 👍

**Review by jhchabran — APPROVED** [review]

**Pull request merged**